usecases/objects/manager.go

```diff
@@ -44,7 +44,7 @@ type schemaManager interface {
 	ReadOnlyClass(name string) *models.Class
 	// AddClassProperty it is upsert operation. it adds properties to a class and updates
 	// existing properties if the merge bool passed true.
-	AddClassProperty(ctx context.Context, principal *models.Principal, class *models.Class, merge bool, prop ...*models.Property) (*models.Class, uint64, error)
+	AddClassProperty(ctx context.Context, principal *models.Principal, class *models.Class, className string, merge bool, prop ...*models.Property) (*models.Class, uint64, error)
```

Comment:
**thought:** isn't className part of class model ?

Reply:
not if the class is nil, eg the given className is not present

Reply:
this
> fixes a panic in schema/{class}/properties: If class does not exist a null pointer was accessed before the nil check

Reply:
**thought**: should the `nil` check not occur earlier then rather than changing the contract of the interface?

Reply:
No, we want the nil check to be after the Authorization check (otherwise it would leak information about class existance) and we have those checks always in the handler

Reply:
Ahhh, great point re: information leakage, nice!